Gems/Atom/RPI/Code/Source/RPI.Public/Material/Material.cpp

```diff
@@ -67,6 +67,20 @@ namespace AZ
             m_materialPipelineData = {};
             m_materialAsset = { &materialAsset, AZ::Data::AssetLoadBehavior::PreLoad };
             ShaderReloadNotificationBus::MultiHandler::BusDisconnect();
+            Data::AssetBus::Handler::BusDisconnect();
+            if (!m_materialAsset.IsReady())
```

Comment:
Just before this, should it check to make sure that the asset ID is valid and return if it's not?

Reply:
Not a bad idea. Will do.